shared/async_macros.rs

```diff
@@ -0,0 +1,68 @@
+macro_rules! wrap_signal_iterator {
```

Comment:
Can I ask for some comment at the top of the file, explaining what it does, why it is in this strange place, the trick with symlinks, etc… once once someone finds it here, they could be a bit surprised.